lib/vrl/stdlib/src/parse_key_value.rs

```diff
@@ -373,14 +375,23 @@ fn parse_value<'a, E: ParseError<&'a str> + ContextError<&'a str>>(
 }
 
 /// Parses the key.
-/// Parsing strategies are the same as parse_value, but we don't need to convert the result to a `Value`.
+/// Overall parsing strategies are the same as parse_value, but we don't need to convert the result to a `Value`.
+/// Standalone key are handle here so quoted standalone key that contains a delimiter will be dealt with correctly.
 fn parse_key<'a, E: ParseError<&'a str> + ContextError<&'a str>>(
     key_value_delimiter: &'a str,
+    field_delimiter: &'a str,
+    standalone_key: bool,
 ) -> impl Fn(&'a str) -> IResult<&'a str, &'a str, E> {
     move |input| {
         alt((
             parse_delimited('"', key_value_delimiter),
-            parse_undelimited(key_value_delimiter),
+            verify(parse_delimited('"', field_delimiter), |_: &str| {
+                standalone_key
+            }),
+            verify(parse_undelimited(key_value_delimiter), |s: &str| {
+                !standalone_key || !s.contains(field_delimiter)
+            }),
+            parse_undelimited(field_delimiter),
```

Comment:
I'll switch that conditional to https://docs.rs/nom/6.1.2/nom/combinator/fn.cond.html